File: jav/palindromo.java
public class palindromo {
    public static boolean EPalindromo(String texto){
        String novo_texto = texto.strip().replace(" " , "").replace("," , "").replace("." , "").replace("-" , "").toLowerCase();
        System.out.println(novo_texto);
        int tamanho_frase = novo_texto.length();
        String novo_texto_reverso = "";
        for (int i = tamanho_frase - 1; i>=0; i--){
            novo_texto_reverso += novo_texto.charAt(i);
        }
        System.out.println(novo_texto);
        System.out.println(novo_texto_reverso);
        if (novo_texto.equals(novo_texto_reverso)){
            System.out.println("A frase '" + texto + "' É um palindromo");
            return true;
        }
        System.out.println("A frase '" + texto + "' NÃO é um palindromo");
        return false;
    }
        public static void main(String[] args) {
            String frase = "A cara rajada da jaararaca";
            boolean resposta = EPalindromo(frase);
        }
}

File: logica-py/palindromo.py
def EPalindromo(texto: str):
    texto_novo = texto.strip().lower().replace(' ','').replace('.','').replace(',','').replace('!','').replace('-','')
    print(texto_novo)
    if texto_novo == texto_novo[::-1]:
        print(f'A frase "{texto}" é um palindromo!')
        return True
    print(f'A frase "{texto}" NÃO é um palindromo!')
    return False


frase = 'A cara rajada da jararaca'
frase2 = 'Socorram-me, subi no onibus em Marrocos!'
EPalindromo(frase)